ui/tests/acceptance/sync/secrets/overview-test.js

```diff
@@ -19,50 +19,94 @@ module('Acceptance | sync | overview', function (hooks) {
-    await click(ts.navLink('Secrets Sync'));
-    await click(ts.destinations.list.create);
-    await click(ts.createCancel);
-    await click(ts.overviewCard.actionLink('Create new'));
-    await click(ts.createCancel);
-    await waitFor(ts.overview.table.actionToggle(0));
-    await click(ts.overview.table.actionToggle(0));
-    await click(ts.overview.table.action('sync'));
-    await click(ts.destinations.sync.cancel);
-    await click(ts.breadcrumbLink('Secrets Sync'));
-    await waitFor(ts.overview.table.actionToggle(0));
-    await click(ts.overview.table.actionToggle(0));
-    await click(ts.overview.table.action('details'));
-    assert.dom(ts.tab('Secrets')).hasClass('active', 'Navigates to secrets view for destination');
+  module('when feature is activated and has pre-existing destinations', function (hooks) {
+    hooks.beforeEach(async function () {
+      syncScenario(this.server);
+    });
+
+    test('it should transition to correct routes when performing actions', async function (assert) {
+      await click(ts.navLink('Secrets Sync'));
+      await click(ts.destinations.list.create);
+      await click(ts.createCancel);
+      await click(ts.overviewCard.actionLink('Create new'));
+      await click(ts.createCancel);
+      await waitFor(ts.overview.table.actionToggle(0));
+      await click(ts.overview.table.actionToggle(0));
+      await click(ts.overview.table.action('sync'));
+      await click(ts.destinations.sync.cancel);
+      await click(ts.breadcrumbLink('Secrets Sync'));
+      await waitFor(ts.overview.table.actionToggle(0));
+      await click(ts.overview.table.actionToggle(0));
+      await click(ts.overview.table.action('details'));
+      assert.dom(ts.tab('Secrets')).hasClass('active', 'Navigates to secrets view for destination');
+    });
   });
 
-  test('it should show opt-in banner and modal if secrets-sync is not activated', async function (assert) {
-    assert.expect(3);
-    this.server.get('/sys/activation-flags', () => {
-      return {
-        data: {
-          activated: [''],
-          unactivated: ['secrets-sync'],
-        },
-      };
+  module('when feature is not activated', function (hooks) {
+    hooks.beforeEach(async function () {
+      let wasActivatePOSTCalled = false;
+
+      // simulate the feature being activated once /secrets-sync/activate has been called
+      this.server.get('/sys/activation-flags', () => {
+        if (wasActivatePOSTCalled) {
+          return {
+            data: {
+              activated: ['secrets-sync'],
+              unactivated: [''],
+            },
+          };
+        } else {
+          return {
+            data: {
+              activated: [''],
+              unactivated: ['secrets-sync'],
+            },
+          };
+        }
+      });
+
+      this.server.post('/sys/activation-flags/secrets-sync/activate', () => {
+        wasActivatePOSTCalled = true;
+        return {};
+      });
```

Comment:
this was a funny hack to implement. my gut says we should fix up the `/activation-flags/` mirage handlers to make actual changes to the schema so that the `get` response is updated after the `post` response is called, but this is good enough for now 😬